source/RoslynAnalyzers/CaseSensitiveEnumParsingAnalyzer.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Octopus.RoslynAnalyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class CaseSensitiveEnumParsingAnalyzer : DiagnosticAnalyzer
+    {
+        const string DiagnosticId = "Octopus_CaseSensitiveEnumParsing";
+
+        const string Title = "Enum.Parse is case sensitive";
+
+        const string MessageFormat = "The Enum.Parse call is case sensitive, which is likely not intended. Use an overload that explicitely defines case behaviour, or our `.ToEnum()` extension method.";
```

Comment:
This may somehow couple this analyzer with our `ToEnum` extension method in other libraries